src/components/TrackPlayer.jsx

```diff
@@ -0,0 +1,32 @@
+import React, { useState } from 'react'
+import { Audio } from 'expo-av'
+import styles from '../styles/styles'
+import { Ionicons } from '@expo/vector-icons'
+
+const TrackPlayer = ({ post }) => {
+  let songURI = post.preview_url
+  const [playback, setPlayback] = useState()
+
+  async function startPlayback() {
+    const playback = new Audio.Sound()
+    await playback.loadAsync({ uri: songURI })
+    await playback.playAsync()
+    setPlayback(playback)
+  }
+  async function stopPlayback() {
+    await playback.stopAsync()
+    setPlayback(undefined)
```

Comment:
What is this line doing? (Curious!)

Reply:
`playback` is actually an object in this case, and can't participate in the `stop`/`play`/`loadAsync()` functions unless it is.  So it can't be `null` or a boolean for the playback functionality to work.